sdk/opendp/smartnoise/evaluation/learner/_Qlearning.py

```diff
@@ -0,0 +1,85 @@
+import numpy as np
+import pandas as pd 
+import csv
+import logging
+
+from opendp.smartnoise.evaluation.params._learner_params import LearnerParams
+from opendp.smartnoise.evaluation.learner._dp_env import DPEnv
+from opendp.smartnoise.evaluation.learner._generate import Grammar
+from opendp.smartnoise.evaluation.learner._computeactions import compute_action
+from opendp.smartnoise.evaluation.learner.util import write_to_csv
+logging.basicConfig(filename="Q-learning.log", level=logging.DEBUG)
+
+class Qlearning():
+    """
+    Use Q-learning to conduct reinforcement learning based query search in evaluator
+    """
+   
+    def Qlearning(self, ep: LearnerParams):       
+        generate query pool
+        with open ("select.cfg", "r") as cfg:
```

Comment:
Qlearning(ep, grammar, query_pool)  # we should move config file reading to input objects, and have the file reading happen outside of the object